Emby.Server.Implementations/Networking/NetworkManager.cs

```diff
@@ -56,70 +283,54 @@ private void OnNetworkChanged()
             NetworkChanged?.Invoke(this, EventArgs.Empty);
         }
 
-        public IPAddress[] GetLocalIpAddresses(bool ignoreVirtualInterface = true)
-        {
-            lock (_localIpAddressSyncLock)
-            {
-                if (_localIpAddresses == null)
-                {
-                    var addresses = GetLocalIpAddressesInternal(ignoreVirtualInterface).ToArray();
-
-                    _localIpAddresses = addresses;
-                }
-
-                return _localIpAddresses;
-            }
-        }
-
-        private List<IPAddress> GetLocalIpAddressesInternal(bool ignoreVirtualInterface)
+        private List<IPAddress> GetLocalIpAddressesInternal()
         {
-            var list = GetIPsDefault(ignoreVirtualInterface).ToList();
+            var list = GetIPsDefault().ToList();
 
             if (list.Count == 0)
             {
                 list = GetLocalIpAddressesFallback().GetAwaiter().GetResult().ToList();
             }
 
-            var listClone = list.ToList();
+            var listClone = new List<IPAddress>();
 
-            return list
-                .OrderBy(i => i.AddressFamily == AddressFamily.InterNetwork ? 0 : 1)
-                .ThenBy(i => listClone.IndexOf(i))
-                .Where(FilterIpAddress)
-                .GroupBy(i => i.ToString())
-                .Select(x => x.First())
-                .ToList();
-        }
+            var subnets = LocalSubnetsFn();
 
-        private static bool FilterIpAddress(IPAddress address)
-        {
-            if (address.IsIPv6LinkLocal
-                || address.ToString().StartsWith("169.", StringComparison.OrdinalIgnoreCase))
+            foreach (var i in list)
             {
-                return false;
-            }
+                if (i.IsIPv6LinkLocal || i.ToString().StartsWith("169.254.", StringComparison.OrdinalIgnoreCase))
+                {
+                    continue;
+                }
 
-            return true;
-        }
+                if (Array.IndexOf(subnets, "[" + i.ToString() + "]") == -1)
```

Comment:
```suggestion
                if (Array.IndexOf(subnets, $"[{i}]") == -1)
```

Reply:
Had to look up $ - hadn't seen its use before.